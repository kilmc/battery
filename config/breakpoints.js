const breakpointsConfig = {
  sm: '600px',
  md: '795px',
  lg: '1025px'
};

export default breakpointsConfig;
