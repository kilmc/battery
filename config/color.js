export const valuesConfig = {
  'green-800': '#098530',
  'green-700': '#05AF3C',
  'green-500': '#25CB68',
  'green-400': '#2BDE73',
  'blue-600': '#003BFF',
  'blue-500': '#2B60FF',
  'grey-900': '#282828',
  'grey-600': '#656868',
  'grey-500': '#9B9E9E',
  'grey-300': '#DCDEDD',
  'grey-200': '#E8E8E8',
  'grey-100': '#FBFBFA',
  'red-500': '#EF0707',
  'red-400': '#FF5151',
};

const propsConfig = [
  {
    propName: "",
    prop: ["color"],
    allowedValues: ["red-500", "grey-900", "green-800"],
    allowedPseudoValues: {
      hover: [],
      active: []
    }
  },
  {
    propName: "bg",
    prop: ["background-color"],
    allowedValues: ["red-500", "grey-900", "green-800"],
    allowedPseudoValues: {
      hover: [],
      active: []
    }
  },
  {
    propName: "border",
    prop: ["border-color"]
  },
  {
    propName: "fill",
    prop: ["fill"]
  },
  {
    propName: "stroke",
    prop: ["stroke"]
  }
];

export default colorPropsConfig;


const propsConfig = {
  lengthUnits: {
    bg: "background-size",
    border: "border-width",
    m: "margin",
    mt: "margin-top",
    mr: "margin-right",
    mb: "margin-bottom",
    ml: "margin-left",
    mx: "margin-left margin-right",
    my: "margin-top margin-bottom",
    p: "padding",
    pt: "padding-top",
    pr: "padding-right",
    pb: "padding-bottom",
    pl: "padding-left",
    h: "height",
    "min-h": "min-height",
    "max-h": "max-height",
    w: "width",
    "min-w": "min-width",
    "max-w": "max-width"
  },
  integers: {
    grow: "flex-grow",
    order: "order",
    shrink: "flex-shrink",
    z: "z-index"
  }
};


