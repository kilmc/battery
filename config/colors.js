const colorsConfig = {
  'green-800': '#098530',
  'green-700': '#05AF3C',
  'green-500': '#25CB68',
  'green-400': '#2BDE73',
  'blue-600': '#003BFF',
  'blue-500': '#2B60FF',
  'grey-900': '#282828',
  'grey-600': '#656868',
  'grey-500': '#9B9E9E',
  'grey-300': '#DCDEDD',
  'grey-200': '#E8E8E8',
  'grey-100': '#FBFBFA',
  'red-500': '#EF0707',
  'red-400': '#FF5151',
};

export default colorsConfig;
