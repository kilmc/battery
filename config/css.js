const cssConfig = {
  pseudoSelectors: {
    hover: {
      pseudo: 'hover',
      pseudoName: 'hover',
      separator: '-',
      prefixOrSuffix: 'prefix'
    },
    focus: {
      pseudo: 'focus',
      pseudoName: 'focus',
      separator: '-',
      prefixOrSuffix: 'prefix'
    }
  }
};

export default cssConfig;
