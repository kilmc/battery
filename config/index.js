import breakpointsConfig from './breakpoints'
import colorsConfig from './colors';
import propsConfig from './props';
import unitsConfig from './units';

const config = {
  breakpoints: breakpointsConfig,
  colors: colorsConfig,
  units: unitsConfig,
  props: propsConfig
};

export default config;
