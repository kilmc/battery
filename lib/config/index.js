'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.processConfig = exports.convertSubProps = exports.formatBorderProp = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) { return typeof obj; } : function (obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; };

var _slicedToArray = function () { function sliceIterator(arr, i) { var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"]) _i["return"](); } finally { if (_d) throw _e; } } return _arr; } return function (arr, i) { if (Array.isArray(arr)) { return arr; } else if (Symbol.iterator in Object(arr)) { return sliceIterator(arr, i); } else { throw new TypeError("Invalid attempt to destructure non-iterable instance"); } }; }();

var _utils = require('../utils');

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var formatBorderProp = exports.formatBorderProp = function formatBorderProp(rootProp, subProp) {
  var _rootProp$split = rootProp.split('-'),
      _rootProp$split2 = _slicedToArray(_rootProp$split, 2),
      start = _rootProp$split2[0],
      end = _rootProp$split2[1];

  return start + '-' + subProp + '-' + end;
};

var convertSubProps = function convertSubProps(config) {
  var propConfigs = config.props;


  var subPropConfigs = propConfigs.filter(function (x) {
    return _typeof(x.subProps) === 'object';
  });

  var convertedPropConfigs = subPropConfigs.map(function (subPropConfig) {
    var prop = subPropConfig.prop,
        propName = subPropConfig.propName,
        subProps = subPropConfig.subProps,
        _subPropConfig$subPro = subPropConfig.subPropSeparator,
        subPropSeparator = _subPropConfig$subPro === undefined ? '' : _subPropConfig$subPro,
        rest = _objectWithoutProperties(subPropConfig, ['prop', 'propName', 'subProps', 'subPropSeparator']);

    return Object.keys(subProps).reduce(function (accumPropConfigs, x) {
      var subProp = subProps[x].split(' ');
      var processedSubProp = prop.match('border') ? subProp.map(function (y) {
        return formatBorderProp(prop, y);
      }).join(' ') : subProp.map(function (y) {
        return prop + '-' + y;
      }).join(' ');

      var newPropConfig = _extends({
        prop: processedSubProp,
        propName: '' + propName + subPropSeparator + x
      }, rest);

      return accumPropConfigs.concat(newPropConfig);
    }, []);
  }).reduce(function (accum, x) {
    return accum.concat(x);
  }, []);

  return _extends({}, config, {
    props: [].concat(_toConsumableArray(config.props), _toConsumableArray(convertedPropConfigs))
  });
};

exports.convertSubProps = convertSubProps;
var processPropConfigs = function processPropConfigs(config) {
  return convertSubProps(config);
};

var processConfig = exports.processConfig = function processConfig(config) {
  return processPropConfigs(config);
};