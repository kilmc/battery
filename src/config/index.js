export const formatBorderProp = (rootProp, subProp) => {
  const [start, end] = rootProp.split('-');
  return `${start}-${subProp}-${end}`;
};

export const convertSubProps = config => {
  const { props: propConfigs } = config;

  const subPropConfigs = propConfigs.filter(
    x => typeof x.subProps === 'object'
  );

  const convertedPropConfigs = subPropConfigs
    .map(subPropConfig => {
      const {
        prop,
        propName,
        subProps,
        subPropSeparator = '',
        ...rest
      } = subPropConfig;

      return Object.keys(subProps).reduce((accumPropConfigs, x) => {
        const subProp = subProps[x].split(' ');
        const processedSubProp = prop.match('border-')
          ? subProp.map(y => formatBorderProp(prop, y)).join(' ')
          : subProp.map(y => `${prop}-${y}`).join(' ');

        const newPropConfig = {
          prop: processedSubProp,
          propName: `${propName}${subPropSeparator}${x}`,
          ...rest
        };

        return accumPropConfigs.concat(newPropConfig);
      }, []);
    })
    .reduce((accum, x) => accum.concat(x), []);

  return {
    ...config,
    props: [...config.props, ...convertedPropConfigs]
  };
};

const processPropConfigs = config => {
  return convertSubProps(config);
};

export const processConfig = config => {
  return processPropConfigs(config);
};
