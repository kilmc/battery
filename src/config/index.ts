import { BatteryConfig } from '../types/';

export const formatBorderProp = (rootProp: string, subProp: string): string => {
  const [start, end] = rootProp.split('-');
  return `${start}-${subProp}-${end}`;
};

export const convertSubProps = (config: BatteryConfig): BatteryConfig => {
  const { props: propConfigs } = config;

  const subPropConfigs = propConfigs.filter(
    x => typeof x.subProps === 'object'
  );

  const convertedPropConfigs = subPropConfigs
    .map(subPropConfig => {
      const {
        prop,
        propName,
        subProps,
        subPropSeparator = '',
        ...rest
      } = subPropConfig;

      return Object.keys(subProps).reduce((accumPropConfigs, x) => {
        const subProp = subProps[x].split(' ');
        const processedSubProp = prop.match('border-')
          ? subProp.map(y => formatBorderProp(prop, y)).join(' ')
          : subProp.map(y => `${prop}-${y}`).join(' ');

        const newPropConfig = {
          prop: processedSubProp,
          propName: `${propName}${subPropSeparator}${x}`,
          ...rest
        };

        return accumPropConfigs.concat(newPropConfig);
      }, []);
    })
    .reduce((accum, x) => accum.concat(x), []);

  return {
    ...config,
    props: [...config.props, ...convertedPropConfigs]
  };
};

const processPropConfigs = (config: BatteryConfig) => {
  return convertSubProps(config);
};

export const processConfig = (config: BatteryConfig) => {
  return processPropConfigs(config);
};
