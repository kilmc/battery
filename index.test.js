/* eslint-env jest, node */

import {
  generate,
  generateColorsRegex,
  generateLengthUnitRegex,
  generateManualClassNameRegex,
  generateRegexes,
} from './index';

import config from './config';

const testConfig = {
  colors: {
    'green-800': '#098530',
    'green-700': '#05AF3C',
    'green-500': '#25CB68'
  },
  props: {
    backgroundSize: {
      prop: 'background-size',
      propName: 'bg',
      enableLengthUnits: true,
      manual: {
        separator: '-',
        values: {
          'cover': 'cover',
          'contain': 'contain',
          'full-height': 'auto 100%',
          'full-width': '100% auto'
        }
      }
    }
  }
};


const propsConfig = {
  'basis': 'flex-basis',
  'mt': 'margin-top'
};

describe('generate', () => {
  expect(generate('mt basis', propsConfig)).toBe('{"mt":{"margin-top":""},"basis":{"flex-basis":""}}');
});

test('generateManualClassNameRegex', () => {
  expect(generateManualClassNameRegex(testConfig.props)).toMatchSnapshot();
});

test('generateLengthUnitRegex', () => {
  expect(generateLengthUnitRegex(testConfig.props)).toMatchSnapshot();
});

test('generateColorsRegex', () => {
  expect(generateColorsRegex(testConfig.colors)).toMatchSnapshot();
});

test('generateRegexes', () => {
  expect(generateRegexes(testConfig)).toMatchSnapshot();
});
