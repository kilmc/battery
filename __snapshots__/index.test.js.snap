// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateColorsRegex 1`] = `
Object {
  "colorsRegex": "(green-800|green-700|green-500)",
}
`;

exports[`generateLengthUnitRegex 1`] = `
Object {
  "lengthUnitRegex": "(bg)(\\\\d+|-\\\\d+)?)",
}
`;

exports[`generateManualClassNameRegex 1`] = `
Object {
  "manualClassNameRegex": "(bg-cover|bg-contain|bg-full-height|bg-full-width)",
}
`;

exports[`generateRegexes 1`] = `
Object {
  "colorsRegex": "(green-800|green-700|green-500)",
  "lengthUnitRegex": "(bg)(\\\\d+|-\\\\d+)?)",
  "manualClassNameRegex": "(bg-cover|bg-contain|bg-full-height|bg-full-width)",
}
`;
